### ISSUE TITLE
after installing this library, my app size increase almost 40MB. Now, total 73MB. Is it normal for this library? or can i reduce app size? thank u!

### ISSUE THREAD

**Issue by saikaungkhant** [HarvestProfit/react-native-rectangle-scanner#64]

**Comment by humphreyja** [comment]
@saikaungkhant Yes it does add a lot unfortunately. Here's why. Apple/iOS has a built in API for edge detection in images which makes the iOS side of this really easy and doesn't need any external packages.  Unfortunately the same is not true for Android. For Android, we need to include the entire OpenCV package with this library. OpenCV's solution to this is garbage too. They recommend using "OpenCV Loader" which requires a user to first install this separate app on their phone before your app can use OpenCV. While this is fine for things like research/college/personal projects, this is not what you want for a polished and sellable end product. Rather you want to just download your app and not need to worry about dependencies. (I don't even understand how Android would allow this, but ok 🤷‍♂️). 

So that's why it adds so much size. Personally, I would really like to see Android come out with an API for edge detection similar to iOS, but I haven't heard anything about this actually happening. So for the time being we are out of luck.

**Comment by saikaungkhant** [comment]
@humphreyja Thank you for your detailed explanation. I want to detect cards so I used rectangle detector. If the scanner detect the rectangle i want to take photo of the card. Is there anything else than this solution that is more light weight. Could you kindly guide again?

**Comment by humphreyja** [comment]
Unfortunately I'd say no. The only solution would be to fork this code, rip out openCV and use the openCV loader from the Android Store. I believe OpenCV's website has documentation on how to do that. The rest of the code should work the same after you've loaded it externally I believe.
